tests/storage/test_background_update.py

```diff
@@ -311,25 +315,25 @@ async def update(progress, count):
 
         # the first update was run with the default batch size, this should be run with minimum batch size
         # as the first items took a very long time
-        async def update(progress, count):
+        async def update_short(progress: JsonDict, count: int) -> int:
             self.assertEqual(progress, {"my_key": 2})
             self.assertEqual(count, 5)
             await self.updates._end_background_update("test_update")
             return count
 
-        self.update_handler.side_effect = update
+        self.update_handler.side_effect = update_short
         self.get_success(self.updates.do_next_background_update(False))
 
 
 class BackgroundUpdateControllerTestCase(unittest.HomeserverTestCase):
-    def prepare(self, reactor, clock, homeserver):
+    def prepare(self, reactor: MemoryReactor, clock: Clock, hs: HomeServer) -> None:
         self.updates: BackgroundUpdater = self.hs.get_datastores().main.db_pool.updates
         # the base test class should have run the real bg updates for us
         self.assertTrue(
             self.get_success(self.updates.has_completed_background_updates())
         )
 
-        self.update_deferred = Deferred()
+        self.update_deferred: Deferred[int] = Deferred()
```

Comment:
I thought we usually had to quote the `Deferred` with a value, but CI seems happy. 🤷

Reply:
I thought so too---IIRC to keep the olddeps job happy. Don't think we've bumped the min twisted version though, so I don't see what's changed to permit this.

Reply:
We discussed in synapse-dev and this isn't needed in the annotations for local variables, because they don't get evaluated (**annotations in function signatures *do***, presumably to attach them to the function object for inspection).

Further, since Python 3.7 (which is all we support), we can now `from __future__ import annotations` and not need to worry about quoting them anywhere, if so desired.